src/util/string.cpp

```diff
@@ -846,13 +851,33 @@ static const std::array<std::wstring, 22> disallowed_dir_names = {
 	L"LPT7",
 	L"LPT8",
 	L"LPT9",
+	L"LPT\u00B2",
+	L"LPT\u00B3",
+	L"LPT\u00B9",
+	L"CONIN$",
+	L"CONOUT$",
 };
 
 /**
  * List of characters that are blacklisted from created directories
  */
 static const std::wstring disallowed_path_chars = L"<>:\"/\\|?*.";
 
+
+/**
+ * @param str
+ * @return A copy of \p str with trailing whitespace removed.
+ */
+static std::wstring wrtrim(const std::wstring &str)
+{
+	size_t back = str.size();
+	while (back > 0 && std::isspace(str[back - 1]))
```

Comment:
doesn't a string have these methods?
```suggestion
	while (!str.empty() && std::isspace(str.back())))
```

Reply:
Or:
```cpp
auto p = str.find_last_not_of(L" \f\n\r\t\v");
return p == str.npos ? "" : str.substr(0, ++p);
```
(But it doesn't respect the locale.)

Reply:
No need to re-invent the wheel. `std::isspace` exists to avoid such hardcoded strings.

Reply:
`back` is changing on each iteration and this code doesn't modify `str` (it's immutable since it's passed as a `const&`), so we can't just use `str.empty()` and `str.back()`.

I'm going to change this to something entirely different though:  leading and trailing ASCII spaces will be replaced with underscores since those cause problems but not other kinds of whitespace according to this: https://support.microsoft.com/en-us/topic/d9813c8e-81f3-ac34-f82e-7db83ade92c2

`\f\n\r\t\v` and other control characters are already blocked.